src/osp/Active/SysRender.h

```diff
@@ -128,8 +162,45 @@ class SysRender
     static void update_delete_groups(
             ACtxRenderGroups &rCtxGroups, IT_T first, IT_T last);
 
+private:
+
+
 }; // class SysRender
 
+void SysRender::add_draw_transforms_recurse(
+        acomp_storage_t<ACompHierarchy> const& hier,
+        acomp_storage_t<ACompDrawTransform>& rDrawTf, ActiveEnt ent)
+{
+    if (rDrawTf.contains(ent))
+    {
+        return;
+    }
+
+    rDrawTf.emplace(ent); // Emplace the component
+
+    ACompHierarchy const &currHier = hier.get(ent);
+
+    if (currHier.m_level <= 1)
+    {
+        return; // Stop recursing when about to reach root
+    }
+
+    // Tail recurse into parent
+    add_draw_transforms_recurse(hier, rDrawTf, currHier.m_parent);
+}
+
+template<typename IT_T>
+void SysRender::assure_draw_transforms(
```

Comment:
consider taking two template args to support C++20's sentinal interators

Reply:
Hmm, just looked this up and I'm already a fan of them. Unfortunately, the subsequent functions used in many other places need to support them too. I'll consider adding this to Longeron++.